mhkit/wave/IO/NDBC/NDBC_request_data.m

```diff
@@ -33,140 +32,88 @@
-            '.YYYY,ndbc_data.ID_' a '.year_' b ...
-            '.MM,ndbc_data.ID_' a '.year_' b '.DD, ndbc_data.ID_' a '.year_' b '.hh, 0,0);']);
+    % Delete compressed and uncompressed files
+    delete(compressed_filename);
+    delete(uncompressed_filename{1});
+
+    % Parse year and station_id from filename
+    [~, filename_no_extension, ~] = fileparts(uncompressed_filename{1});
+    if contains(filename_no_extension, '_')
+        % Handles filenames like: 42002w2008_old.txt.gz
+        filename_base = extractBefore(filename_no_extension, '_');
+    else
+        filename_base = filename_no_extension;
     end
-   end
- 
-else
-    for i=1:length(key)
-        sha=cell(datapd{string(py.str(key{i}))}.values.shape);
-        x=int64(sha{1,1});
-        y=int64(sha{1,2});
-        df = reshape(double(py.array.array('d',py.numpy.nditer(datapd{string(py.str(key{i}))}.values,...
-    pyargs("flags",{"refs_ok"})))),[x,y]);
-        cols = cell(py.list(py.numpy.nditer(datapd{string(py.str(key{i}))}.columns,pyargs("flags",{"refs_ok"}))));
-        
-        for k = 1:length(cols)
-            
-            a = char(string(py.str(key{i})));
-            b = char(string(py.str(cols{k})));
-            c = df(:,k);
-            
-            if parameter == "swden"
-                
-                if ~contains(b, ".")
-                    if contains(b, "YY")
-                        
-                        b = char("YYYY");
-                        
-                    end
-                    eval(['ndbc_data.year_' a '.' b '= c ;']);
-                    
-                else
-                    freq = [freq str2double(b)];
-                    
-                    spectra = [spectra ;c];
-                    
-                end
-            else
-               if contains(b, "YY")
-                   
-                   b = char("YYYY");
-               end
-               eval(['ndbc_data.year_' a '.' b '= c ;']);
-            end
-            
-        end
-        if parameter == "swden"
-            si = size(c);
-            si2= size(freq);
-            spectra = reshape(spectra, [si(1),si2(2)]);
-            freq = freq';
-            spectra = spectra';
-            eval(['ndbc_data.year_' a '.frequency = freq ;']);
-            eval(['ndbc_data.year_' a '.spectrum = spectra ;']);
-        end
-        freq = [];
-        spectra = [];
-        
-        eval(['ndbc_data.year_' a '.time = datetime(ndbc_data.year_' a '.YYYY,ndbc_data.year_' a ...
-            '.MM,ndbc_data.year_' a '.DD, ndbc_data.year_' a '.hh,0,0);']);
+    year = str2double(extractAfter( ...
+        filename_base, ...
+        strlength(filename_base) - 4));     % grab last 4 chars
+    station_id = extractBefore( ...
+        filename_base, ...
+        strlength(filename_base) - 5 + 1);  % grab all but last 5 chars
+
+    % Rename year column and add datetime column to table
+    file_table.Properties.VariableNames{'YY'} = 'YYYY';
```

Comment:
I am getting "Unrecognized table variable name 'YY' in the environmental contours example notebook. Looking back at the previous version of the code, this conversion was an if statement, we likely need to reinstate the 'if' statement.

Reply:
Resolved in commit c9b2174

Reply:
Issues with associated live scripts (notebooks) are resolved in commit b1b2e70